### PR TITLE
Rework config into dotenv

### PR DESCRIPTION
**Pull request by frederickfogerty**
This change allows users to set environment variables which makes it easier to set up on Docker, Heroku, AWS, etc, and easier to test.

### CODE REVIEW

**Comment by megawac** [comment]
I like this
/cc @jimaek you dig?

**Comment by frederickfogerty** [comment]
Before this is merged we need to make sure the trusted users works, as obviously you guys use it a lot. I don't use it so I didn't test it, but everything else works

File: settings.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+config['owner'] = os.environ.get("OWNER")
+
+config['user'] = os.environ.get("USER")
+config['auth_token'] = os.environ.get("AUTH_TOKEN")
+config['post_secret'] = os.environ.get("HOOK_SECRET")
+config['merge_re'] = os.environ.get("MERGE_REGEX")
+config['path_to_repo'] = os.environ.get("REPO_PATH")
+config['repo_branch'] = os.environ.get("REPO_BRANCH")
+config['repo_remote'] = os.environ.get("REPO_REMOTE")
+config['trusted_users'] = ['holisbot']
```
Comment (megawac) [inline settings.py]: Shouldn't this be `os.environ.get("TRUSTED_USERS").split(",")`

Comment (frederickfogerty) [inline settings.py]: I was literally editing that just now haha

File: validator.py [hunk 2]
```diff
@@ -7,9 +7,10 @@
 
 from collections import deque
 
-class PullBot(PullValidator, GitMerger):
-    config = yaml.load(open("config.yml"))
+from settings import config
 
+class PullBot(PullValidator, GitMerger):
+    config = config
```
Comment (megawac) [inline validator.py]: why is this line here?

Comment (frederickfogerty) [inline validator.py]: Other files, such as `utils/merger.py`, access `self.config` and I didn't want to restructure that in case you guys liked it that way. (I'm not a python programmer so I don't know what the best practices are).

Comment (megawac) [inline validator.py]: oh right, gotcha. Haven't worked on this repo in a while

**Comment by frederickfogerty** [comment]
Closing due to becoming stale
